scene/gui/text_edit.cpp

```diff
@@ -463,17 +463,22 @@ void TextEdit::_notification(int p_what) {
 
 		case NOTIFICATION_INTERNAL_PHYSICS_PROCESS: {
 			if (scrolling && get_v_scroll() != target_v_scroll) {
-				double target_y = target_v_scroll - get_v_scroll();
-				double dist = abs(target_y);
-				// To ensure minimap is responsive override the speed setting.
-				double vel = ((target_y / dist) * ((minimap_clicked) ? 3000 : v_scroll_speed)) * get_physics_process_delta_time();
+				double delta_y = target_v_scroll - get_v_scroll();
+				double dist = abs(delta_y);
+				// To ensure minimap is responsive, override the speed setting.
+				double vel = SIGN(delta_y) * (minimap_clicked ? 3000 : v_scroll_speed) * get_physics_process_delta_time();
+
+				// Speed up scrolling when the target line is relatively far away (won't be reached in 8 physics frames.)
+				if (abs(vel) < dist / 8.0) {
```

Comment:
This value should likely be computed from a constant time, not a constant number of frames, as the physics process FPS can be modified by users.

Reply:
Doned, I think. I'm not sure how to test it, I'm kind of unfamiliar with the TextEdit node itself, but my change in general doesn't seem to affect it. Not sure why.